### PR TITLE
GenAI: warn when there are unsaved ModelCard changes

### PR DESCRIPTION
**Pull request by Erin007**
## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/alert/Alert.tsx [hunk 1]
```diff
@@ -12,6 +12,14 @@ import moduleStyles from './alert.module.scss';
...
 type AlertType = 'primary' | 'success' | 'danger' | 'warning' | 'info' | 'gray';
 
+export const alertTypes: {[key in AlertType]: AlertType} = {
+  primary: 'primary',
+  success: 'success',
+  danger: 'danger',
+  warning: 'warning',
+  info: 'info',
+  gray: 'gray',
+};
```
Comment (Erin007) [inline apps/src/componentLibrary/alert/Alert.tsx:22]: @levadadenys I followed the pattern for `buttonColors` What do you think?

File: apps/src/code-studio/components/progress/LessonProgress.jsx [hunk 2]
```diff
@@ -175,13 +185,44 @@ class LessonProgress extends Component {
...
+               * When the user clicks on a level bubble for which we support changing to
+               * that level without reloading the page, we use a click handler which
+               * calls through to the progress redux store to change to that level
+               * immediately. Unless we're on a GenAi level with unsaved changes, in which
+               * case we show a reminder to explicitly save the changes.
+               **/
+              const getOnBubbleClick = (level, currentLevel) => {
+                if (isGenAiLevel() && hasUnsavedChanges()) {
+                  console.log('show save reminder should be the function');
+                  return setShowSaveChangesReminder(true);
```
Comment (Erin007) [inline apps/src/code-studio/components/progress/LessonProgress.jsx]: @sanchitmalhotra126 This is my messy WIP with a few different paths I tried. I think the thing that gets us closest to the Product/Design ask is for the bubble click to trigger us to display the unsaved changes warning, but I feel like I'm still mixing functionality that shouldn't be mixed. Could use a 🦆 please. 

a.) Is this even the best way to do this? 
b.) if it is the way to accomplish what we want, these components handle dispatch differently - do you have advice for how to work around that?

Comment (sanchitmalhotra126) [reply apps/src/code-studio/components/progress/LessonProgress.jsx]: Yeah this is super tricky for all the reasons you mentioned 🙂 there are so many paths to change levels and the signal that lab2 uses to listen for level change is just that the current level ID has updated. So doing anything that 1) knows when a level is changing and 2) also _prevents_ the level change from happening is difficult until and unless we have some singular central level switching system. And yep agreed that it's not ideal to have Gen AI (or any lab) - specific logic in the level progress layer.

Let's discuss at the Gen AI sync, but I think in the short term, it doesn't seem super feasible for us to block level changing in lab2. Longer term, we potentially could have a central level switching controller that all level change requests are routed through, and individual labs/other components could register handlers to respond to and optionally block level changes. But for now if it's alright for us to just show a banner/warning to users, that would be much more straightforward to implement.

**Comment by Erin007** [comment]
closing in favor of #60585

**Pull request closed without merging**